### PR TITLE
[extension/sigv4authextension] Add Unmarshal() to sigv4auth extension (PR)

### PR DESCRIPTION
**Pull request by erichsueh3**
**Description:** <Describe what has changed.>
This PR addresses an issue brought up in #16490 and refactors the code in `extension/sigv4authextension/config.go`.

**Link to tracking Issue:** #16490

### CODE REVIEW

**Comment by erichsueh3** [comment]
Also note that the changelog entry should be skipped as this is code refactoring without any functional change

**Comment by erichsueh3** [comment]
Hi @djaglowski anything else I can do to get this PR merged?

**Comment by erichsueh3** [comment]
> Hi @djaglowski anything else I can do to get this PR merged?

Hi @djaglowski, just wanted to follow up on this again

**Comment by erichsueh3** [comment]
> > Hi @djaglowski anything else I can do to get this PR merged?
> 
> Hi @djaglowski, just wanted to follow up on this again

Hi @djaglowski, wanted to check in here again, what can I do to get this merged? It's a small refactor PR that shouldn't cause too much trouble.

File: extension/sigv4authextension/config.go [hunk 1]
```diff
@@ -41,10 +42,18 @@ type AssumeRole struct {
 // compile time check that the Config struct satisfies the component.Config interface
 var _ component.Config = (*Config)(nil)
 
-// Validate checks that the configuration is valid.
```
Comment (atoulme) [inline extension/sigv4authextension/config.go:44]: why did you remove that doc?

**Comment by atoulme** [comment]
@erichsueh3 please rebase and a changelog entry under .chloggen, copying the TEMPLATE entry.
Please run `make lint` over your changes as well.

**Comment by erichsueh3** [comment]
> @erichsueh3 please rebase and a changelog entry under .chloggen, copying the TEMPLATE entry. Please run `make lint` over your changes as well.

Hey @atoulme, I've rebased, added a changelog entry, and ran `make lint` as well, let me know if anything else needs to be done

File: extension/sigv4authextension/config.go (hunk 1 above)
Comment (erichsueh3) [reply extension/sigv4authextension/config.go:42]: Now that Validate() effectively isn't doing anything, I removed the comment because it wasn't entirely accurate - I can re-add it though, if this is just a consistency thing

File: .chloggen/sigv4_add_unmarshal.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: sigv4authextension
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add Unmarshal() method to config.go
```
Comment (atoulme) [inline .chloggen/sigv4_add_unmarshal.yaml:8]: Can we get a bit more information in this changelog? If this just an internal change, just remove this changelog and mark the PR as "Skip Changelog"

File: extension/sigv4authextension/config.go (hunk 1 above)
Comment (atoulme) [reply extension/sigv4authextension/config.go:42]: Validate is still needed looks like, since these errors can be returned:
```
	if err != nil {
		return fmt.Errorf("could not retrieve credential provider: %w", err)
	}
	if credsProvider == nil {
		return fmt.Errorf("credsProvider cannot be nil")
	}
```

My 0.02c: move the checks to a Validate method, and keep this unmarshal method to initialize the config properly.
